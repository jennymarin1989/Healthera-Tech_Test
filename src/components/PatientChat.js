import React, { Component } from 'react';
import SubmitButton from './SubmitButton';



class PatientChat extends Component {
		constructor(){
			super();
			this.state ={
				message: "Message",
				messages: []
			}
			this.updatePatientMessage = this.updatePatientMessage.bind(this)
		}
		updatePatientMessage = (event) =>{
			this.setState({
				message: event.target.value
			});
		}

    render(){
        return(
            <div className="container">
            <div className="row">
                <div className="col-4">
                    <div className="card">
                        <div className="card-body">
                            <div className="card-title">Messages</div>
                            <hr/>
                            <div className="messages">
														
                            </div>
                        </div>
                        <div className="card-footer">
                                <input type="text" placeholder="Message" value={this.state.message} onChange={this.updatePatientMessage} className="form-control"/>
                                <br/>
                                <SubmitButton
																	name={"Send"}
																	id={"send-msg"}
																/>
                        </div>
                    </div>
                </div>
            </div>
        </div>

        )
    }

}

export default PatientChat;