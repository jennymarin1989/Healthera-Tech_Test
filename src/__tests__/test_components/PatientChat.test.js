import React from 'react';
import { shallow } from 'enzyme';
import PatientChat from '../../components/PatientChat';


describe('PatientChat',() => {

		let chat = shallow(<PatientChat />)
	
		it("renders correctly", () => {
			expect(chat).toMatchSnapshot();
		});	

	describe("initializes state succesfully", () => {
		it("initializes state for message and messages", () => {
			expect(chat.state('message')).toEqual("");
		})
	})	

})