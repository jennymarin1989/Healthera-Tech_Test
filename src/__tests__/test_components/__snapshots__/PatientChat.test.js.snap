// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientChat renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PatientChat />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <div
                className="col-4"
        >
                <div
                        className="card"
                >
                        <div
                                className="card-body"
                        >
                                <div
                                        className="card-title"
                                >
                                        Messages
                                </div>
                                <hr />
                                <div
                                        className="messages"
                                />
                        </div>
                        <div
                                className="card-footer"
                        >
                                <input
                                        className="form-control"
                                        onChange={[Function]}
                                        placeholder="Message"
                                        type="text"
                                        value="Message"
                                />
                                <br />
                                <SubmitButton
                                        id="send-msg"
                                        name="Send"
                                />
                        </div>
                </div>
        </div>
</div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-4"
>
          <div
                    className="card"
          >
                    <div
                              className="card-body"
                    >
                              <div
                                        className="card-title"
                              >
                                        Messages
                              </div>
                              <hr />
                              <div
                                        className="messages"
                              />
                    </div>
                    <div
                              className="card-footer"
                    >
                              <input
                                        className="form-control"
                                        onChange={[Function]}
                                        placeholder="Message"
                                        type="text"
                                        value="Message"
                              />
                              <br />
                              <SubmitButton
                                        id="send-msg"
                                        name="Send"
                              />
                    </div>
          </div>
</div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card"
>
            <div
                        className="card-body"
            >
                        <div
                                    className="card-title"
                        >
                                    Messages
                        </div>
                        <hr />
                        <div
                                    className="messages"
                        />
            </div>
            <div
                        className="card-footer"
            >
                        <input
                                    className="form-control"
                                    onChange={[Function]}
                                    placeholder="Message"
                                    type="text"
                                    value="Message"
                        />
                        <br />
                        <SubmitButton
                                    id="send-msg"
                                    name="Send"
                        />
            </div>
</div>,
          "className": "col-4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-body"
>
                <div
                                className="card-title"
                >
                                Messages
                </div>
                <hr />
                <div
                                className="messages"
                />
</div>,
              <div
                className="card-footer"
>
                <input
                                className="form-control"
                                onChange={[Function]}
                                placeholder="Message"
                                type="text"
                                value="Message"
                />
                <br />
                <SubmitButton
                                id="send-msg"
                                name="Send"
                />
</div>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card-title"
>
                    Messages
</div>,
                  <hr />,
                  <div
                    className="messages"
/>,
                ],
                "className": "card-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Messages",
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": "Messages",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "messages",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Message"
                    type="text"
                    value="Message"
/>,
                  <br />,
                  <SubmitButton
                    id="send-msg"
                    name="Send"
/>,
                ],
                "className": "card-footer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "onChange": [Function],
                    "placeholder": "Message",
                    "type": "text",
                    "value": "Message",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "id": "send-msg",
                    "name": "Send",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-4"
          >
                    <div
                              className="card"
                    >
                              <div
                                        className="card-body"
                              >
                                        <div
                                                  className="card-title"
                                        >
                                                  Messages
                                        </div>
                                        <hr />
                                        <div
                                                  className="messages"
                                        />
                              </div>
                              <div
                                        className="card-footer"
                              >
                                        <input
                                                  className="form-control"
                                                  onChange={[Function]}
                                                  placeholder="Message"
                                                  type="text"
                                                  value="Message"
                                        />
                                        <br />
                                        <SubmitButton
                                                  id="send-msg"
                                                  name="Send"
                                        />
                              </div>
                    </div>
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-4"
>
            <div
                        className="card"
            >
                        <div
                                    className="card-body"
                        >
                                    <div
                                                className="card-title"
                                    >
                                                Messages
                                    </div>
                                    <hr />
                                    <div
                                                className="messages"
                                    />
                        </div>
                        <div
                                    className="card-footer"
                        >
                                    <input
                                                className="form-control"
                                                onChange={[Function]}
                                                placeholder="Message"
                                                type="text"
                                                value="Message"
                                    />
                                    <br />
                                    <SubmitButton
                                                id="send-msg"
                                                name="Send"
                                    />
                        </div>
            </div>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card"
>
              <div
                            className="card-body"
              >
                            <div
                                          className="card-title"
                            >
                                          Messages
                            </div>
                            <hr />
                            <div
                                          className="messages"
                            />
              </div>
              <div
                            className="card-footer"
              >
                            <input
                                          className="form-control"
                                          onChange={[Function]}
                                          placeholder="Message"
                                          type="text"
                                          value="Message"
                            />
                            <br />
                            <SubmitButton
                                          id="send-msg"
                                          name="Send"
                            />
              </div>
</div>,
            "className": "col-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-body"
>
                  <div
                                    className="card-title"
                  >
                                    Messages
                  </div>
                  <hr />
                  <div
                                    className="messages"
                  />
</div>,
                <div
                  className="card-footer"
>
                  <input
                                    className="form-control"
                                    onChange={[Function]}
                                    placeholder="Message"
                                    type="text"
                                    value="Message"
                  />
                  <br />
                  <SubmitButton
                                    id="send-msg"
                                    name="Send"
                  />
</div>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="card-title"
>
                      Messages
</div>,
                    <hr />,
                    <div
                      className="messages"
/>,
                  ],
                  "className": "card-body",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Messages",
                      "className": "card-title",
                    },
                    "ref": null,
                    "rendered": "Messages",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "messages",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      className="form-control"
                      onChange={[Function]}
                      placeholder="Message"
                      type="text"
                      value="Message"
/>,
                    <br />,
                    <SubmitButton
                      id="send-msg"
                      name="Send"
/>,
                  ],
                  "className": "card-footer",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "onChange": [Function],
                      "placeholder": "Message",
                      "type": "text",
                      "value": "Message",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "id": "send-msg",
                      "name": "Send",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
